### PR TITLE
pywal16: init at 3.5.3

### PR DESCRIPTION
**Pull request by MyGitHubBlueberry**
## Description of changes
This is fork of pywal, named [pywal16](https://github.com/eylles/pywal16), maintained by [eylles](https://github.com/eylles).
This package generates color palletes based on wallpaper, but it does it in 16 and not 8 color palette.
Some packages support pywal16 already, as original pywal replacement [wpgtk](https://github.com/deviantfero/wpgtk) will be a good example.

Motivation:
- This package is quite nice, I would use it myself
- Fixes pywal 8 color problem
- eylles, the package developer, is building handy [git repo](https://github.com/eylles/pywal16-libadwaita) whith a templates for gui apps, and as person who uses wallust (pywal alternative), I would love to switch to something with gtk-qt support.

Note:
- I build package on my nixos with nix flakes and it works all right. Idk if it is "applicable" testing, so I won't tick this box. 
- I have read contributing.md, but I'm not sure if I did everything ok, so this box is also unticked.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Qyriad** [comment]
Thank you for this, @MyGitHubBlueberry — could you change the commit message and PR title to match the [convention for new packages](https://github.com/NixOS/nixpkgs/tree/master/pkgs#commit-conventions)? ofborg is very picky and only understands commits of a specific format.

File: pkgs/development/python-modules/pywal16/default.nix
Comment (Qyriad) [inline pkgs/development/python-modules/pywal16/default.nix:17]:
```suggestion
{ lib,
  buildPythonPackage,
  fetchPypi,
  imagemagick,
  swww, #feh
  ...
}:

buildPythonPackage rec {
  pname = "pywal16";
  version = "3.5.3";
  format = "setuptools";
```

Packages in `python-modules` should get called as `pkgs.python3.pkgs.callPackage`, and thus should take anything from the `python3.pkgs` scope as arguments directly. You can look at the other pywal package for an example: https://github.com/NixOS/nixpkgs/blob/b7d845292c304e026d86097e6d07409070e80dcc/pkgs/development/python-modules/pywal/default.nix

**Review by Qyriad — CHANGES_REQUESTED** [review]
This also has to be added to [pkgs/top-level/python-packages.nix](pkgs/top-level/python-packages.nix) or it won't actually be accessible in Nixpkgs. i.e., checking out this PR as is, `nix-build -A python3Packages.pywal16` does not work.

**Comment by MyGitHubBlueberry** [comment]
> Thank you for this, @MyGitHubBlueberry — could you change the commit message and PR title to match the [convention for new packages](https://github.com/NixOS/nixpkgs/tree/master/pkgs#commit-conventions)? ofborg is very picky and only understands commits of a specific format.

On it.

**Comment by MyGitHubBlueberry** [comment]
@Qyriad, thank you, all done.

**Review by Qyriad — CHANGES_REQUESTED** [review]
This still needs an entry in [pkgs/top-level/python-packages.nix](pkgs/top-level/python-packages.nix) or it won't actually be accessible in Nixpkgs. If I check out this PR right now, I can't actually build the package with something like `nix-build -A python3Packages.pywal16` or `nix build '.#python3Packages.pywal16'`

**Comment by MyGitHubBlueberry** [comment]
@Qyriad sorry, I'm not very comfortable with git and github and I only now noticed that you actually suggested changes and not only description how to make them.
Again, thank you :)

**Comment by MyGitHubBlueberry** [comment]
@Qyriad is there anything else I need to do?

**Review by Qyriad — APPROVED** [review]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pywal16</li>
    <li>python311Packages.pywal16.dist</li>
    <li>python312Packages.pywal16</li>
    <li>python312Packages.pywal16.dist</li>
  </ul>
</details>

Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pywal16</li>
    <li>python311Packages.pywal16.dist</li>
    <li>python312Packages.pywal16</li>
    <li>python312Packages.pywal16.dist</li>
  </ul>
</details>

**Comment by Qyriad** [comment]
LGTM, sorry for the wait

**Comment by MyGitHubBlueberry** [comment]
@Qyriad there is no rash, thank you for your time.
You sent reviews, so can I tick the box `aarch64-darwin`?
And still, do I need to do anything, or should I just wait for the package to be merged?

**Comment by Qyriad** [comment]
Those checkboxes (I think) are for things you did. There is now at least one approval though, so hopefully a committer will be able to look at this PR soon

**Review by MyGitHubBlueberry — COMMENTED** [review]
Thank you for you review :)

**Comment by wegank** [comment]
I'm not sure what's going on here, but the package was added in #316805 in a different way, so this PR can't be merged. Please add the patches in another PR if they are necessary.

**Pull request closed without merging**
